Release executables for Windows, MacOS and Linux and automation on pushing new changes

marco97pa opened the pull request:
Added a Workflow to build executables on pushing new changes

Created a new GitHub Workflow that automates the building of new executables for each release. It will output the binaries for Windows, MacOS and Linux in the releases folder.

Whenever an update to the code is pushed the new executables will be available within minutes in the releases folder.

Also added a short description of the releases in the README.md

marco97pa committed:
Added a Workflow to build executables on push

Created a new GitHub Workflow that automates the building of new executables for each release whenever new code is pushed. It will output the binaries for Windows, MacOS and Linux in the releases folder.

Also added a short description of the releases in the README.md

marco97pa committed:
Solved a typo in the filename in the build process

marco97pa committed:
Changed last step of the workflow to commit to the repo

marco97pa committed:
Changed last step to commit to the repository

marco97pa committed:
New release of app-ubuntu-latest

marco97pa committed:
Limited the number of parrallel jobs to avoid conflicting commits

marco97pa committed:
New release of app-macos-latest

marco97pa committed:
Added .gitignore and improved the pushing process of the Workflow

marco97pa committed:
New release of app-macos-latest

marco97pa committed:
New release of app-ubuntu-latest

marco97pa committed:
New release of app-windows-latest

marco97pa committed:
Cleanup

marco97pa committed:
New release of app-macos-latest

marco97pa committed:
New release of app-ubuntu-latest

marco97pa committed:
New release of app-windows-latest

wetcork commented:
Ciao!
Innanzitutto grazie per il contributo, non sono esperto di GitHub actions, non le ho mai usate, quindi chiedo direttamente a te, potresti fare una modifica in modo che l'output delle build sia _raiplay-(sistema operativo)-latest_?

marco97pa commented:
certo, posso modificarlo facilmente.
Quale formato preferisci tra:  
- raiplay-dl-{sistema_operativo}-latest
- raiplay-{sistema_operativo}-latest 
?  
Personalmente preferisco il primo, volendo potrei anche rimuovere "-latest". 

Oltre questa modifica, correggerò anche il nome della cartella di output: al momento si chiama "dist" ma nel Readme e nei commit l'ho chiamata "releases" che secondo me è un nome più azzeccato e intuitivo

wetcork commented:
In realtà credo di preferire semplicemente _raiplay-{sistema_operativo}-latest_ per evitare di mettere un trattino in più.
Il nome della cartella perfetto.

marco97pa committed:
Changed executables names to raiplay-{os}-latest
Also improved documentation and removed some unnecessary files generated by the building process

marco97pa committed:
New release of raiplay-macos-latest

marco97pa committed:
New release of raiplay-ubuntu-latest

marco97pa committed:
New release of raiplay-windows-latest

marco97pa commented:
Completate le modifiche richieste, ripuliti alcuni file inutili generati durante il processo di building e come puoi vedere il bot di GitHub Action ha aggiunto gli eseguibili con il nuovo naming.
